### PR TITLE
support metadata for micro

### PR DESCRIPTION
**Pull request by ripienaar**

### CODE REVIEW

**Comment by ripienaar** [comment]
Implements https://github.com/nats-io/nats-architecture-and-design/pull/205

File: micro/service.go [hunk 1]
```diff
@@ -90,9 +91,10 @@ type (
 
 	// ServiceIdentity contains fields helping to identity a service instance.
 	ServiceIdentity struct {
-		Name    string `json:"name"`
-		ID      string `json:"id"`
-		Version string `json:"version"`
+		Name     string            `json:"name"`
+		ID       string            `json:"id"`
+		Version  string            `json:"version"`
+		Metadata map[string]string `json:"metadata"`
```
Comment (piotrpio) [inline micro/service.go:97]: Maybe we could use `omitempty` for metadata to avoid sending `null` field if it's not provided?

Comment (ripienaar) [reply micro/service.go:97]: updated it in both places.

Comment (aricart) [reply micro/service.go:97]: (I actually believe the same)

**Comment by ripienaar** [comment]
@piotrpio tests green now, pls merge when happy.

**Pull request merged**